spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -565,7 +564,10 @@ private void sendFileToRemoteDirectory(InputStream inputStream, String temporary
 		String tempRemoteFilePath = temporaryRemoteDirectory + fileName;
 		// write remote file first with temporary file extension if enabled
 
-		String tempFilePath = tempRemoteFilePath + (this.useTemporaryFileName ? this.temporaryFileSuffix : "");
+		String tempFilePath = tempRemoteFilePath;
+		if (!FileExistsMode.APPEND.equals(mode)) {
+			tempFilePath += this.useTemporaryFileName ? this.temporaryFileSuffix : "";
```

Comment:
Why do we need to append empty string to the end if not `this.useTemporaryFileName`?
How about this instead:
```
if (!FileExistsMode.APPEND.equals(mode) &&  this.useTemporaryFileName) {
			tempFilePath += this.temporaryFileSuffix;
}
```